clang/test/CodeGenSYCL/amd-address-space-conversions.cpp

```diff
@@ -1,128 +1,128 @@
 // RUN: %clang_cc1 -triple amdgcn-amd-amdhsa -fsycl-is-device -disable-llvm-passes -emit-llvm %s -o - | FileCheck %s
 void bar(int &Data) {}
-// CHECK: define dso_local void @[[RAW_REF:[a-zA-Z0-9_]+]](ptr noundef nonnull align 4 dereferenceable(4) %
+// CHECK-DAG: define {{.*}} void @[[RAW_REF:[a-zA-Z0-9_]+]](ptr noundef nonnull align 4 dereferenceable(4) %
```

Comment:
We could actually check the calling convention and linkage instead of simply skipping it.

Reply:
I see this in place: https://github.com/intel/llvm/commit/1ea43e2a77667eab3d8702732afe62cf9f6347a9, and followed the pattern established here to help avoid downstream conflicts. 
Actually checking the calling convention and linkage makes sense too.